### PR TITLE
抽离脚本引擎 + 添加 BTN 的脚本支持

### PR DESCRIPTION
**Pull request by Ghost-chu**
## Release Notes

- **New Features**
	- Integrated a script engine for dynamic rule evaluation.
	- Added support for script-based peer banning rules.
	- Introduced new messaging keys for enhanced clarity in error and status messages related to the rule engine.

- **Bug Fixes**
	- Improved error handling and logging during script compilation and execution.

- **Documentation**
	- Updated documentation to reflect changes in method signatures and new features related to script handling.

- **Chores**
	- Refactored existing code to accommodate the new script engine and improve maintainability.

### CODE REVIEW

**Comment by paulzzh** [comment]
@coderabbitai review

File: src/main/java/com/ghostchu/peerbanhelper/module/RuleFeatureModule.java [hunk 1]
```diff
@@ -5,19 +5,20 @@
...
+import java.util.List;
 import java.util.concurrent.ExecutorService;
 
 public interface RuleFeatureModule extends FeatureModule {
     /**
      * 检查一个特定的 Torrent 和 Peer 是否应该封禁
      *
      * @param torrent             Torrent
-     * @param peer                Peer
+     * @param peers                Peers
```
[the remark this answers is not in the document]
Comment (Ghost-chu) [reply src/main/java/com/ghostchu/peerbanhelper/module/RuleFeatureModule.java:16]: Generate commitable sugggestion

File: src/main/java/com/ghostchu/peerbanhelper/scriptengine/ScriptEngine.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                    threadSafe,
+                    version,
+                    scriptContent,
+                    expression
+            );
+        } catch (IOException e) {
+            log.warn("Script Engine unable to compile the script: {}", fallbackName);
+            return null;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (Ghost-chu) [reply src/main/java/com/ghostchu/peerbanhelper/scriptengine/ScriptEngine.java:156]: Generate committable suggestion

File: src/main/java/com/ghostchu/peerbanhelper/module/impl/rule/BtnNetworkOnline.java [hunk 3]
```diff
@@ -155,9 +165,77 @@ public boolean isThreadSafe() {
...
+                    if (expressionRun.action() == PeerAction.BAN) {
+                        if (checkResult.get().action() != PeerAction.SKIP) {
+                            checkResult.set(expressionRun);
+                        }
+                    }
+                });
+            }
+        }
+        return checkResult.get();
+
```
[the remark this answers is not in the document]
Comment (Ghost-chu) [reply src/main/java/com/ghostchu/peerbanhelper/module/impl/rule/BtnNetworkOnline.java:206]: Generate committable suggestion

**Comment by Ghost-chu** [comment]
@coderabbitai review

**Pull request merged**